core/domain/acl_decorators.py

```diff
@@ -1051,3 +1074,43 @@ def test_can_manage_topic_rights(self, topic_id, **kwargs):
     test_can_manage_topic_rights.__wrapped__ = True
 
     return test_can_manage_topic_rights
+
+
+def can_accept_suggestion(handler):
+    """Decorator to check if user can accept the given suggestion."""
+
+    @can_edit_exploration
+    def test_can_accept_suggestion_to_exploration(unused_self, unused_exp_id):
+        pass
```

Comment:
a bit weird but do we plan to modify this as part of milestone2 or is this just a placeholder for adding the can_edit_exploration decorator?

Reply:
This was in face weird. It was just going to be a placeholder 😅 . I discussed it with Yogesh and he had an idea that we can have a function that checks the "can_edit_exploration" permissions and we can call it here (but I am not sure how we should implement this because after checking exceptions need to be raised and all exceptions are defined in baseHandler).

I am not entirely sure of what best to do here. I thought about it and here is an approach I thought of.

- So we restructure our URL to /generalsuggestionactionhandler/exploration/<exp_id>/<suggestion_id> for suggestions to explorations (similarly /topic/<topic_id>/<suggestion_id> for topics -- a different controller as such will be used). We will need one controller per type of target being suggested to.
- For the exploration suggestions we will use @can_edit_exploration on the handler and on topic suggestions we will use @can_edit_topic and so on
- We could use multiple decorators on the same handler (I didn't know this!). So we can create a separate decorator which checks with the scoring and use both of these decorators on each handler. 

Does this sound good? @seanlip @anmolshkl? Thanks!

Reply:
If you're using separate URLs then calling them /generalsuggestionactionhandler seems a bit weird to me. Perhaps just /suggestionhandler/.../.../... may be better.

The other points seem reasonable to me, though it may be helpful to see the code that results. Try and simultaneously come up with a brief playbook along the lines of "to add a new suggestion type, create a URL of this form, use decorator XXX, etc." to make sure that what you're doing here is well-defined and generalizable. Write this in a doc somewhere, and link to it; this will be useful context for reviews.

Reply:
Actually the general prefix is to distinguish from /suggestionactionhandler (the existing URL for suggestion framework). Once we migrate and remove old controllers, we will drop the prefix 'general'. 

Yeah I'll update the playbook with these instructions too! I'll make this change in a bit.

Reply:
Ah ok (re the prefix). Just make sure to add a TODO in the code for that.

Reply:
Sure will add a TODO! Thanks :)